likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/detail/view/LMFeedPostDetailFragment.kt

```diff
@@ -0,0 +1,1403 @@
+                    }
+
+                    else -> {
+                        //input text is a comment
+                        addComment(postId, updatedText)
+                    }
+                }
+                LMFeedViewUtils.hideKeyboard(this)
+                etComment.text = null
+            }
+            setCommentSendButton(false)
+
+            setRemoveReplyingToClickListener {
+                hideReplyingToView()
+            }
+        }
+    }
+
+    // adds the comment locally and calls api
+    private fun addComment(postId: String, updatedText: String) {
+        binding.rvPostDetails.apply {
+            val createdAt = System.currentTimeMillis()
+            val tempId = "-${createdAt}"
+
+            // calls api
+            postDetailViewModel.addComment(postId, tempId, updatedText)
+
+            // adds comment locally
+            val commentViewData = postDetailViewModel.getCommentViewDataForLocalHandling(
+                postId,
+                createdAt,
+                tempId,
+                updatedText,
+                null
+            )
+
+            // remove NoCommentsViewData if visible
+            if (getItem(commentsCountPosition) is LMFeedNoCommentsViewData) {
+                removeItem(commentsCountPosition)
+            }
+            // gets old [CommentsCountViewData] from adapter
+            if (getItem(commentsCountPosition) != null) {
+                val oldCommentsCountViewData =
+                    (getItem(commentsCountPosition) as LMFeedCommentsCountViewData)
+
+                // updates old [CommentsCountViewData] by adding to [commentsCount]
+                val updatedCommentsCountViewData = oldCommentsCountViewData.toBuilder()
+                    .commentsCount(oldCommentsCountViewData.commentsCount + 1)
+                    .build()
+
+                //updates [CommentsCountViewData]
+                updateItem(commentsCountPosition, updatedCommentsCountViewData)
+            } else {
+                // creates new [CommentsCountViewData] when the added comment is first
+                val newCommentsCountViewData = LMFeedCommentsCountViewData.Builder()
+                    .commentsCount(1)
+                    .build()
+                addItem(commentsCountPosition, newCommentsCountViewData)
+            }
+
+            // gets post from adapter
+            var post = getItem(postDataPosition) as LMFeedPostViewData
+
+            //update the footer view data
+            val updatedFooterView = post.footerViewData.toBuilder()
+                .commentsCount(post.footerViewData.commentsCount + 1)
+                .build()
+
+            //updated the post
+            post = post.toBuilder()
+                .footerViewData(updatedFooterView)
+                .build()
+
+            // notifies the subscribers about the change in post data
+            //todo:
+//            postEvent.notify(Pair(post.id, post))
```

Comment:
please confirm whether we have added this or not.

Reply:
added in further PRs